product_multi_company/migrations/17.0.1.1.0/pre-migrate.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2025 Moduon Team S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0)
+from logging import getLogger
+
+from openupgradelib import openupgrade
+
+from odoo.tools import safe_eval
+
+_logger = getLogger(__name__)
+
+
+PREVIOUS_DOMAIN = [
+    "|",
+    ("company_ids", "in", "COMPANY_IDS"),
+    ("company_ids", "=", False),
+]
+
+
+UPSTREAM_DOMAIN = " [('company_id', 'in', company_ids + [False])]"
```

Comment:
Also update this to match the upstream domain in 18.0 https://github.com/odoo/odoo/blob/d4886faf12ccaf63d5e899c20df2543d1ce046ab/addons/product/security/product_security.xml#L37